rmgpy/rmg/react.py

```diff
@@ -104,6 +104,112 @@ def reactMolecules(moleculeTuples):
     return reactionList
 
 def deflate(rxns, molecules, reactantIndices):
+def findDegeneracies(rxnList, useSpeciesReaction = True):
+    """
+    given a list of reaction object with Molecule objects, this method 
+    removes degenerate reactions and
+    increments the degeneracy of the reaction object. This method modifies
+    rxnList in place and does not return anything.
+
+    This algorithm used to exist in family.__generateReactions, but was moved
+    here because it didn't have any family dependence.
+    """
+    # These duplicates should be combined and the reaction degeneracy should be increased
+    # Reaction degeneracy is only increased if two reaction are isomorphic but resulted
+    # from different transition states.
+    # We want to sort all the reactions into sublists composed of isomorphic reactions
+    rxnSorted = []
+    for rxn0 in rxnList:
+        # find resonance structures for rxn0
+        convertToSpeciesObjects(rxn0)
+        if len(rxnSorted) == 0:
+            # This is the first reaction, so create a new sublist
+            rxnSorted.append([rxn0])
+        else:
+            # Loop through each sublist, which represents a unique reaction
+            for rxnList1 in rxnSorted:
+                # Try to determine if the current rxn0 is identical or isomorphic to any reactions in the sublist
+                isomorphic = False
+                identical = False
+                sameTemplate = False
+                for rxn in rxnList1:
+                    isomorphic = rxn0.isIsomorphic(rxn,checkIdentical=False)
+                    identical = rxn0.isIsomorphic(rxn,checkIdentical=True)
```

Comment:
Great idea! This would be an awesome speedup. Added this in a squash commit (for the second case under the assumption that `not isomorphic` is more common than `is identical`)